lib/resources/kernel_module.rb

```diff
@@ -36,19 +68,51 @@ def loaded?
       !found.nil?
     end
 
-    def version
-      if inspec.os.redhat? || inspec.os.name == 'fedora'
-        modinfo_cmd = "/sbin/modinfo -F version #{@module}"
-      else
-        modinfo_cmd = "modinfo -F version #{@module}"
-      end
+    def disabled?
+      !modprobe_output.match(%r{^install\s+#{@module}\s+/(s?)bin/(true|false)}).nil?
+    end
 
-      cmd = inspec.command(modinfo_cmd)
+    def disabled_via_bin_true?
+      !modprobe_output.match(%r{^install\s+#{@module}\s+/(s?)bin/true}).nil?
+    end
+
+    def disabled_via_bin_false?
+      !modprobe_output.match(%r{^install\s+#{@module}\s+/(s?)bin/false}).nil?
+    end
+
+    def blacklisted?
```

Comment:
So I am returning a true of false in each of those, thus if I put the result of each of them into an array and then checked if the array contained true then return true else false. 

That makes sense?